benchmarks/src/utils.rs

```diff
@@ -0,0 +1,204 @@
+            _auxiliary_data_inputs,
+            created_notes,
+        ) = mock_inputs_with_existing(
+            MockAccountType::StandardExisting,
+            AssetPreservationStatus::Preserved,
+            account,
+            input_notes,
+        );
+        let output_notes = created_notes.into_iter().filter_map(|note| match note {
+            OutputNote::Public(note) => Some(note),
+            OutputNote::Private(_) => None,
+        });
+        let mut tx_args = TransactionArgs::default();
+        tx_args.extend_expected_output_notes(output_notes);
+
+        Self {
+            account,
+            block_header,
+            block_chain,
+            notes: consumed_notes,
+            tx_args,
+        }
+    }
+
+    pub fn tx_args(&self) -> &TransactionArgs {
+        &self.tx_args
+    }
+}
+
+impl Default for MockDataStore {
+    fn default() -> Self {
+        Self::new(AssetPreservationStatus::Preserved)
+    }
+}
+
+impl DataStore for MockDataStore {
+    fn get_transaction_inputs(
+        &self,
+        account_id: AccountId,
+        block_num: u32,
+        notes: &[NoteId],
+    ) -> Result<TransactionInputs, DataStoreError> {
+        assert_eq!(account_id, self.account.id());
+        assert_eq!(block_num, self.block_header.block_num());
+        assert_eq!(notes.len(), self.notes.len());
+
+        let notes = self
+            .notes
+            .iter()
+            .filter(|note| notes.contains(&note.id()))
+            .cloned()
+            .collect::<Vec<_>>();
+
+        Ok(TransactionInputs::new(
+            self.account.clone(),
+            None,
+            self.block_header,
+            self.block_chain.clone(),
+            InputNotes::new(notes).unwrap(),
+        )
+        .unwrap())
+    }
+
+    fn get_account_code(&self, account_id: AccountId) -> Result<ModuleAst, DataStoreError> {
+        assert_eq!(account_id, self.account.id());
+        Ok(self.account.code().module().clone())
+    }
+}
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+pub fn get_account_with_default_account_code(
+    account_id: AccountId,
+    public_key: Word,
+    assets: Option<Asset>,
+) -> Account {
+    let account_code_src = DEFAULT_ACCOUNT_CODE;
+    let account_code_ast = ModuleAst::parse(account_code_src).unwrap();
+    let account_assembler = TransactionKernel::assembler();
+
+    let account_code = AccountCode::new(account_code_ast.clone(), &account_assembler).unwrap();
+    let account_storage = AccountStorage::new(vec![SlotItem {
+        index: 0,
+        slot: StorageSlot::new_value(public_key),
+    }])
+    .unwrap();
+
+    let account_vault = match assets {
+        Some(asset) => AssetVault::new(&[asset]).unwrap(),
+        None => AssetVault::new(&[]).unwrap(),
+    };
+
+    Account::new(account_id, account_vault, account_storage, account_code, Felt::new(1))
+}
+
+#[cfg(feature = "std")]
```

Comment:
Related to one of the previous comments: if we make the crate always `std` we can get rid of this (and other similar) feature gates.